### PR TITLE
Add ShuffledDataset' class to shuffle datasets, add slimpajama 5m dataset for 70B training

### PR DESCRIPTION
**Pull request by lessw2020**
<not ready for review> 
This PR adds
1 - A shuffled dataset to wrap the current hf dataset and provide us with shuffling.  This is to see if that resolves the large spikes we get in training loss.  

2 -  the slimpajama-6B dataset to our dataset options.
This dataset has 5.5M entries and is 14GB. 
Thus it hopefully represents a good middle ground for where minipile is a bit too small for larger models like 70B but c4 is proving to be way too prone to network issues at scale (since it must be streamed). 

It is 1%, evenly sampled of the slimpajama-627B dataset and has elements from c4 and roughly 9 other datasets.

### CODE REVIEW

File: torchtrain/datasets/hf_datasets.py [hunk 1]
```diff
@@ -148,6 +167,30 @@ def __iter__(self):
                     f"being re-looped. Loss related metrics might be misleading.{Color.reset}"
                 )
 
+class ShuffleDataset(IterableDataset):
+  def __init__(self, dataset, buffer_size):
```
Comment (wanchaol) [inline torchtrain/datasets/hf_datasets.py:171]: I think `shuffle` can just be a kwarg flag of HuggingFaceDataset, and if it's true then we do random operation, instead of introducing a separate class wrapper?
